python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -0,0 +1,597 @@
+        self.config.doCrosstalkCoeffs = False
+        self.config.doTransmissionCurve = True
+
+
+class MockLSSTDataContainer(object):
+    """Container for holding ISR LSST mock objects.
+    """
+    dataId = "isrMockLSST Fake Data"
+    darkval = 2.  # e-/sec
+    oscan = 250.  # DN
+    gradient = .10
+    exptime = 15.0  # seconds
+    darkexptime = 15.0  # seconds
+
+    def __init__(self, **kwargs):
+        if 'config' in kwargs.keys():
+            self.config = kwargs['config']
+        else:
+            self.config = None
+
+    def expectImage(self):
+        if self.config is None:
+            self.config = IsrMockLSSTConfig()
+        self.config.doGenerateImage = True
+        self.config.doGenerateData = False
+
+    def expectData(self):
+        if self.config is None:
+            self.config = IsrMockLSSTConfig()
+        self.config.doGenerateImage = False
+        self.config.doGenerateData = True
+
+    def get(self, dataType, **kwargs):
+        """Return an appropriate data product.
+
+        Parameters
+        ----------
+        dataType : `str`
+            Type of data product to return.
+
+        Returns
+        -------
+        mock : IsrMockLSST.run() result
+            The output product.
+        """
+        if "_filename" in dataType:
+            self.expectData()
+            return tempfile.mktemp(), "mock"
+        elif 'transmission_' in dataType:
+            self.expectData()
+            return TransmissionMockLSST(config=self.config).run()
+        elif dataType == 'ccdExposureId':
+            self.expectData()
+            return 20090913
+        elif dataType == 'camera':
+            self.expectData()
+            return IsrMockLSST(config=self.config).getCamera()
+        elif dataType == 'raw':
+            self.expectImage()
+            return RawMockLSST(config=self.config).run()
+        elif dataType == 'bias':
+            self.expectImage()
+            return BiasMockLSST(config=self.config).run()
+        elif dataType == 'dark':
+            self.expectImage()
+            return DarkMockLSST(config=self.config).run()
+        elif dataType == 'flat':
+            self.expectImage()
+            return FlatMockLSST(config=self.config).run()
+        elif dataType == 'fringe':
+            self.expectImage()
+            return FringeMockLSST(config=self.config).run()
+        elif dataType == 'defects':
+            self.expectData()
+            return DefectMockLSST(config=self.config).run()
+        elif dataType == 'bfKernel':
+            self.expectData()
+            return BfKernelMockLSST(config=self.config).run()
+        elif dataType == 'linearizer':
+            return None
+        elif dataType == 'crosstalkSources':
+            return None
+        else:
+            raise RuntimeError("ISR DataRefMock cannot return %s.", dataType)
+
+
+class MockFringeLSSTContainer(object):
+    """Container for mock fringe data.
+    """
+    dataId = "isrMockLSST Fake Data"
```

Comment:
Same here, but this entire object is probably not useful until fringing is working better.